docs/date.md

```diff
@@ -1,96 +1,398 @@
 # Temporal.Date
 
-A representation of a calendar date.
+A `Temporal.Date` represents a calendar date.
+"Calendar date" refers to the concept of a date as expressed in everyday usage, independent of any time zone.
+For example, it could be used to represent an event on a calendar which happens during the whole day no matter which time zone it's happening in.
+
+`Temporal.Date` refers to the whole of a specific day; if you need to refer to a specific time on that day, use `Temporal.DateTime`.
+A `Temporal.Date` can be converted into a `Temporal.DateTime` by combining it with a `Temporal.Time` using the `withTime()` method.
+
+`Temporal.YearMonth` and `Temporal.MonthDay` carry less information than `Temporal.Date` and should be used when complete information is not required.
 
 ## Constructor
 
 ### **new Temporal.Date**(_year_: number, _month_: number, _day_: number, _disambiguation_: 'constrain' | 'balance' | 'reject' = 'constrain') : Temporal.Date
 
-Creates a new `Date` object that represents a calendar date
+**Parameters:**
+- `year` (number): A year, ranging between &minus;271821 and 275760 inclusive.
+- `month` (number): A month, ranging between 1 and 12 inclusive.
+- `day` (number): A day of the month, ranging between 1 and 31 inclusive.
+- `disambiguation` (optional string): How to deal with out-of-range values of the other parameters.
+  Allowed values are `constrain`, `balance`, and `reject`.
+  The default is `constrain`.
+
+**Returns:** a new `Temporal.Date` object.
+
+Use this constructor if you have the correct parameters for the date already as individual number values, or you need the disambiguation behaviour.
+Otherwise, `Temporal.Date.from()`, which accepts more kinds of input, is probably more convenient.
+
+The `disambiguation` parameter works as follows:
+- In `constrain` mode (the default), any out-of-range values are clamped to the nearest in-range value.
+- In `balance` mode, any out-of-range values are resolved by balancing them with the next highest unit.
+- In `reject` mode, the presence of out-of-range values will cause the constructor to throw a `RangeError`.
+
+Usage examples:
+```javascript
+// Pi day in 2020
+date = new Temporal.Date(2020, 3, 14)  // => 2020-03-14
+
+// Different disambiguation modes
+date = new Temporal.Date(2001, 13, 1, 'constrain')  // => 2001-12-01
+date = new Temporal.Date(2001, -1, 1, 'constrain')  // => 2001-01-01
+date = new Temporal.Date(2001, 13, 1, 'balance')  // => 2002-01-01
+date = new Temporal.Date(2001, -1, 1, 'balance')  // => 2000-11-01
+date = new Temporal.Date(2001, 13, 1, 'reject')  // throws
+date = new Temporal.Date(2001, -1, 1, 'reject')  // throws
+```
 
 ## Static methods
 
 ### Temporal.Date.**from**(_thing_: string | object) : Temporal.Date
 
-### Temporal.Date.**compare**(_one_: Temporal.Date, _two_: Temporal.Date) : number
+**Parameters:**
+- `thing` (string or object): The value representing the desired date.
```

Comment:
Do we want to give more detail here on what the string should/should not be?
Is it just an ISO 8601 string? the output from Date() also?

Reply:
Currently the text says "string must be in ISO 8601 format" and that's what the polyfill accepts. The spec is unclear until we solve #312 and #198. I'm not overly excited about accepting the output of Date.toString() since according to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toString it was implementation-dependent before ES2018, and I can also affect what is printed by changing my `LANG` or `LC_ALL` environment variables.

I opened https://github.com/tc39/proposal-temporal/issues/370 for this.